fetch.bs

```diff
@@ -6710,6 +6748,362 @@ agent's <a>CORS-preflight cache</a> for which there is a <a>cache entry match</a
+      on permissions policy. This would only be the case if the subframe and its parent do not share
+      quota, as quota is either shared or delegated.
+
+      <ol>
+       <li><p>Let <var>otherContainer</var> be <var>otherNavigable</var>'s
+       <a>navigable container</a>.
+
+       <li>
+        <p>If <var>otherContainer</var>'s <a>node document</a> and <var>otherDocument</var> do
+        not <a>share deferred-fetch quota</a>, then:
+
+        <ol>
+         <li><p>If <var>otherContainer</var>'s <a>subframe reserved deferred-fetch quota</a> is
+         <a>optional subframe deferred-fetch quota</a>, and <var>otherDocument</var> is
+         <a>allowed to use</a> the <a>policy-controlled feature</a>
+         {{PermissionPolicy/"deferred-fetch"}}, then increment <var>quota</var> by
+         <a>optional subframe deferred-fetch quota</a>.
+
+         <li><p>Otherwise, if <var>otherContainer</var>'s
+         <a>subframe reserved deferred-fetch quota</a> is
+         <a>minimal subframe deferred-fetch quota</a> and <var>otherDocument</var>
+         is <a>allowed to use</a> the <a>policy-controlled feature</a>
+         {{PermissionPolicy/"deferred-fetch-minimal"}}, then increment <var>quota</var> by
+         <a>minimal subframe deferred-fetch quota</a>.
+        </ol>
+      </ol>
+
+     <li>
+      <p><a for=list>For each</a> <a>deferred fetch record</a> <var>deferredRecord</var> of
+      <var>otherDocument</var>'s <a for=fetch>fetch group</a>'s
+      <a for="fetch group">deferred fetch records</a>:</p>
+      <ol>
+       <li><p>Let <var>requestLength</var> be the <a>total request length</a> of
+       <var>deferredRecord</var>'s <a for="deferred fetch record">request</a>.
+
+       <li><p>Decrement <var>quota</var> by <var>requestLength</var>.
+
+       <li><p>If <var>deferredRecord</var>'s <a for="deferred fetch record">request</a>'s
+       <a for=request>URL</a>'s <a for=url>origin</a> is <a>same origin</a> with <var>origin</var>,
+       then decrement <var>quotaForOrigin</var> by <var>requestLength</var>.
+      </ol>
+    </ol>
+
+   <li>
+    <p>Otherwise, if <var>otherNavigable</var> is not a <a for=/>top-level traversable</a>, and its
+    <a for=navigable>parent</a>'s <a>active document</a> and <var>document</var>
+    <a>share deferred-fetch quota</a>, then decrement <var>quota</var> by
+    <var>otherNavigable</var>'s <a>navigable container</a>'s
+    <a>subframe reserved deferred-fetch quota</a>.
+
+    <p class=note>This is a cross-origin/cross-agent subframe, so it spends some of the quota of
+    this document and its relatives.
+  </ol>
+ <li><p>If <var>quota</var> is less than zero, than return zero.
+
+ <li><p>If <var>quota</var> is less than <var>quotaForOrigin</var>, then
+ return <var>quota</var>.
+
+ <li><p>Return <var>quotaForOrigin</var>.
+</ol>
+</div>
+
+<div algorithm>
+<p>To <dfn export>reserve deferred-fetch quota</dfn> for a <a>navigable container</a>
```

Comment:
```
To reserve deferred-fetch quota for a navigable container container given an origin originToNavigateTo:
```

What will be the returned value given the following example?

```
root (a.com) -> frame_a (a.com) -> frame_c (b.com)
             -> frame_d (b.com) -> frame_b (a.com)
```

- `frame_a`: 64kb
- `frame_b`: should also be 64kb?
- `frame_c`: 8kb?
- `frame_d`: 8kb

Reply:
> if the iframe navigates internally to a different origin

How about initial loads of both frame?

Also, I think there are some issue about the `inherited policy` for `frame_b`. My test shows that it won't have `deferred-fetch` enabled ... Is it because its parent iframe is cross-origin?

Reply:
For initial loads, `frame_a` would receive 64kb as "reserved", but that value would never be read, because the `availalble quota` algorithm only reads that number for frames that don't share quota with their parent.
Ah yes, `frame_b` would have 0. Sorry

Reply:
Is it `0` for `frame_b` (not 8kb)? "deferred-fetch-minimal" is still enabled for it, so it will go through the steps the same as `frame_d`?

Reply:
Yes, but its parent doesn't share quota with the top level, so it can't receive 8kb. It doesn't pass the 3rd condition.

Reply:
Now I think I am still not quite understand the for-loop ...

```
For each navigable of container’s node navigable’s top-level traversable’s inclusive descendant navigables who meets all of the following conditions:
```

Does this basically iterate through all the nodes (iframes + root from this example) in the frame tree?

If so, `frame_c` and `frame_d` will both pass all the conditions?

Reply:
You're right, I've reworded it. Look at the above condition also

Reply:
To summarize:
```
root (a.com) -> frame_a (a.com) -> frame_c (b.com)
                      -> frame_d (b.com) -> frame_b (a.com)
             
```

- `frame_a` will receive 64kb, but it would be ignored because it shares quota with root
- `frame_c` will receive 8kb because it passes all the conditions for minimal quota
- `frame_d` as well
- `frame_b` will receive 0, because it fails the outer condition in (2): (the parent has to share quota with the top levle)